src/compas_fab/robots/semantics.py

```diff
@@ -60,10 +60,13 @@ def __get_group_link_names(self, group):
                     link_names.append(name)
         for subgroup in group.findall('group'):
             if subgroup.attrib['name'] != group.attrib['name']:
-                subgroup_link_names = self.__get_group_link_names(subgroup)
-                for name in subgroup_link_names:
-                    if name not in link_names:
-                        link_names.append(name)
+                # find group element at top level
+                for top_group_elem in self.root.findall('group'):
```

Comment:
This is a lot of nesting. If possible, it'd be good to flatten this a bit.

Reply:
changed in 1fa3689dd7ec6012e2faa6f0440e2cd94981bd0f